### DISCUSSION TITLE
Support for openthread send_udp till raspi socat

### DISCUSSION THREAD

**Discussion by wassfila** [HomeSmartMesh/sdk-hsm-sensortag#5] [Q&A] [answered]
continuation from here 
https://github.com/nRFMesh/nRF52_Mesh/discussions/12

**Comment by wassfila** [comment]
To listen to the UDP port on the raspberry pi, I used socat tool as explained here
https://www.homesmartmesh.com/docs/networks/thread/#udp-test
I did not document how to install it as the package has the same name as the tool
here an example ``sudo apt install socat`
https://www.howtoinstall.me/ubuntu/18-04/socat/

**Comment by tcpipchip** [comment]
Hi, crossing the net sea! 

Well, the correct is grafana😂

**Reply by wassfila** [reply]
Oh I see, well Grafana is another story. Before that, you have to ensure that your UDP packets that you observed on wireshark, arrives to the raspberry pi IPV6 network. For that you should listen with socat like described here
https://www.homesmartmesh.com/docs/networks/thread/#udp-test

And you should see an output of the text content that was created by you uC, in my case it was a text like show here 
https://www.homesmartmesh.com/docs/networks/thread/#thread-udp-packets
`thread_tags/7009D837C7BB557A{"alive":18673,"voltage":3.106,"light":184.948,"temperature":24.56,"humidity":46.49,"pressure":961.90}`

Once you see that on rapi command line output of socat, the rest is going to be easy, I'll give you then more detailed instructions, let's go step by step

**Comment by tcpipchip** [comment]
Hi
Yes!
Reached 4242! :) I can see the UDP reaching it with Wireshark
But the 4242 already in use, then someone is listening! Some web server! I had imagined grafano😅

**Reply by wassfila** [reply]
Do you get a message like this ?
<img width="767" alt="image" src="https://user-images.githubusercontent.com/1510763/158027045-54a6b3da-9e0f-4d4e-a805-6686168b05fa.png">
can you post a screenshot or text log ?

You can figure it out with special command for ipv6 (not for ipv6 most things are different than ipv4)
`sudo netstat -tulpen`
I made a test when I run a listener on purpose and you can see it

<img width="737" alt="image" src="https://user-images.githubusercontent.com/1510763/158027175-2b60d92a-4633-46bc-a67a-7703eb1a0935.png">

grafana is on port :3000 so not likely
then you can of course figure out the path from pid
`pwdx 15309`

**Comment by tcpipchip** [comment]
yes, 
i got
![image](https://user-images.githubusercontent.com/11488460/158027448-61ab9430-d94d-4416-b480-243efcbdefa4.png)
ok, i will test soon on 3000

**Reply by wassfila** [reply]
I don't understand what you mean by `i will test soon on 3000`. Grafana is using port 3000 and someone else `not Grafana` is using the port 4242, it is very likely to be the python scripts from the raspi install, because I selected the port 4242, it is very unlikely that another app used it, so once you figure out the path with pwdx command, you can go stop it and try again the socat as it is on the port 4242.

**Reply by tcpipchip** [reply]
Nice! 

wassfila 

one question, the threads (no mesh), to enter in network, must to be in NCP range ?

**Reply by wassfila** [reply]
yes, must be in NCP range in case that is the only node in the network. And to be more specific, every node must be within range of a router node. Some nodes can be end devices only, those cannot help others to join. So at then we can also say that Thread even with a single NCP is technically still a mesh but of only one node.

**Comment by tcpipchip** [comment]
@wassfila 

I was thinking, imagine 280  end-devices transmitting data to a server by RADIO

Looks that use THREAD on end-devices to transmit using UDP is faster that the server making a BLE SERVER scanning of end-devices making broadcasting...

Right ?

**Reply by wassfila** [reply]
@tcpipchip 
As this is a different question that the scope of this discussion, I shifted it here #7 . It does not matter if we have many small discussions, they're even better organized and easier to find that way :)

**Comment by wassfila** [comment, marked as answer]
This support ticket can be closed as you reached the UDP level with Thread.
Please feel free to open any new discussion, even if it is only to report status and give update. I just want to keep every discussion scoped and in case discussion deviate, then open a new discussion.

The answer to this discussion originally posted by @tcpipchip in #7 
> Hi, making more tests, now node-red listening 3 UDP packages, no CRASH
> 
> See at end of blog https://nina-b3-openthread.blogspot.com/2022/03/nina-b302-transmitindo-dados-via-udp.html
